sources/models/src/lib.rs

```diff
@@ -530,9 +530,28 @@ struct OciDefaults {
 
 ///// The hard and soft limit values for an OCI defaults resource limit.
 #[model(add_option = false)]
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, Eq, Ord, PartialOrd, PartialEq)]
 struct OciDefaultsResourceLimit {
-    hard_limit: u32,
-    soft_limit: u32,
+    #[serde(deserialize_with = "deserialize_limit")]
+    hard_limit: i64,
+    #[serde(deserialize_with = "deserialize_limit")]
+    soft_limit: i64,
+}
+
+impl OciDefaultsResourceLimit {
+    pub fn get_hard_limit(self) -> u64 {
+        match self.hard_limit {
+            -1 => u64::MAX,
+            _ => self.hard_limit as u64,
+        }
+    }
```

Comment:
nit: the body could be factored into a private `get_limit` function that takes either `self.hard_limit` or `self.soft_limit` as a parameter